static/js/components/dashboard/CourseDescription_test.js

```diff
@@ -63,6 +227,44 @@ describe('CourseDescription', () => {
     assert.lengthOf(elements.titleLink, 0);
   });
 
+  it('show view course on edX link when non-enroll statues and enrollment_url is set', () => {
+    const EXPECTED_STATUSES = [
+      STATUS_OFFERED, STATUS_PENDING_ENROLLMENT, STATUS_WILL_ATTEND
+    ];
+
+    for (let status of EXPECTED_STATUSES) {
+      let course = findAndCloneCourse(course => (
+        course.runs.length > 0 &&
+        course.runs[0].status === status
+      ));
+      let firstRun = course.runs[0];
+      firstRun.enrollment_url = 'http://example.com';
+      const wrapper = shallow(<CourseDescription courseRun={firstRun} courseTitle={course.title} />);
+      let elements = getElements(wrapper);
+      assert.equal(elements.titleLink.text(), 'View on edX');
+      assert.isAbove(elements.titleLink.props().href.length, 0);
+      assert.include(elements.titleLink.props().href, firstRun.enrollment_url);
+    }
+  });
+
+  it('does not show view course on edX link when non-enroll statues and enrollment_url is not set', () => {
```

Comment:
please change to: 'does not show an edX link for an unenrolled course run with no enrollment url'